platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramEnqueuer.java

```diff
@@ -34,7 +34,8 @@ private void enqueueExecution(AGUuid programUuid) {
 		AGProgram program = AGProgram.loadOrInsert(programUuid);
 		if (program.reloadForUpdate()) {
```

Comment:
Locking should be performed on the `AGProgramState` record instead, e.g.:
```
var state = AGProgram.loadOrInsert(programUuid).getState();
if (state.reloadForUpdate()) {
	state//
		.setQueuedAt(...)
		.setQueuedBy(...)
		.save();
}
```

Reply:
I changed that.